doc/reference/ndarray.rst

```diff
@@ -12,3 +12,352 @@ For the basic concept of ``ndarray``\s, please refer to the `NumPy documentation
+   dpnp.ndarray.choose
+   dpnp.ndarray.sort
+   dpnp.ndarray.argsort
+   dpnp.ndarray.partition
+   dpnp.ndarray.argpartition
+   dpnp.ndarray.searchsorted
+   dpnp.ndarray.nonzero
+   dpnp.ndarray.compress
+   dpnp.ndarray.diagonal
+
+
+Calculation
+-----------
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   dpnp.ndarray.max
+   dpnp.ndarray.argmax
+   dpnp.ndarray.min
+   dpnp.ndarray.argmin
+   dpnp.ndarray.ptp
+   dpnp.ndarray.clip
+   dpnp.ndarray.conj
+   dpnp.ndarray.conjugate
+   dpnp.ndarray.round
+   dpnp.ndarray.trace
+   dpnp.ndarray.sum
+   dpnp.ndarray.cumsum
+   dpnp.ndarray.mean
+   dpnp.ndarray.var
+   dpnp.ndarray.std
+   dpnp.ndarray.prod
+   dpnp.ndarray.cumprod
+   dpnp.ndarray.all
+   dpnp.ndarray.any
+
+
+Arithmetic, matrix multiplication, and comparison operations
+------------------------------------------------------------
+
+Arithmetic and comparison operations on :class:`dpnp.ndarrays <dpnp.ndarray>`
+are defined as element-wise operations, and generally yield
+:class:`dpnp.ndarray` objects as results.
+
+Each of the arithmetic operations (``+``, ``-``, ``*``, ``/``, ``//``,
+``%``, ``divmod()``, ``**`` or ``pow()``, ``<<``, ``>>``, ``&``,
+``^``, ``|``, ``~``) and the comparisons (``==``, ``<``, ``>``,
+``<=``, ``>=``, ``!=``) is equivalent to the corresponding
+universal function (or **ufunc** for short) in DPNP.  For
+more information, see the section on :ref:`Universal Functions
+<ufunc>`.
+
+
+Comparison operators:
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   dpnp.ndarray.__lt__
+   dpnp.ndarray.__le__
+   dpnp.ndarray.__gt__
+   dpnp.ndarray.__ge__
+   dpnp.ndarray.__eq__
+   dpnp.ndarray.__ne__
+
+
+Unary operations:
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   dpnp.ndarray.__neg__
+   dpnp.ndarray.__pos__
+   dpnp.ndarray.__abs__
+   dpnp.ndarray.__invert__
+
+
+Arithmetic:
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   dpnp.ndarray.__add__
+   dpnp.ndarray.__sub__
+   dpnp.ndarray.__mul__
+   dpnp.ndarray.__truediv__
+   dpnp.ndarray.__floordiv__
+   dpnp.ndarray.__mod__
+   dpnp.ndarray.__divmod__
+   dpnp.ndarray.__pow__
+   dpnp.ndarray.__lshift__
+   dpnp.ndarray.__rshift__
+   dpnp.ndarray.__and__
+   dpnp.ndarray.__or__
+   dpnp.ndarray.__xor__
+
+
+Arithmetic, in-place:
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   dpnp.ndarray.__iadd__
+   dpnp.ndarray.__isub__
+   dpnp.ndarray.__imul__
+   dpnp.ndarray.__itruediv__
+   dpnp.ndarray.__ifloordiv__
+   dpnp.ndarray.__imod__
+   dpnp.ndarray.__ipow__
+   dpnp.ndarray.__ilshift__
+   dpnp.ndarray.__irshift__
+   dpnp.ndarray.__iand__
+   dpnp.ndarray.__ior__
+   dpnp.ndarray.__ixor__
+
+
+Special methods
+---------------
+
+For standard library functions:
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   dpnp.ndarray.__copy__
+   dpnp.ndarray.__deepcopy__
+   dpnp.ndarray.__reduce__
+   dpnp.ndarray.__setstate__
```

Comment:
I do not think `cupy.ndarray` exposes these methods. What is the rationale for their support in `dpnp.ndarray`?

Reply:
These lines were added for future implementations and shouldn`t have been displayed in dpnp documentation.
I'll fix it. Thanks